runner/ollamarunner/runner.go

```diff
@@ -130,16 +130,21 @@ func (s *Server) NewSequence(prompt string, images []llm.ImageData, params NewSe
 	// Ensure that at least 1 input can be discarded during shift
 	params.numKeep = min(params.numKeep, s.cache.numCtx-1)
 
-	if int32(len(inputs)) > s.cache.numCtx {
-		discard := int32(len(inputs)) - s.cache.numCtx
+	// Basic runner-side guard: if truncation is enabled and num_ctx <= 1, return an error
+	if params.truncate && s.cache.numCtx <= 1 {
```

Comment:
Is this check needed as part of this PR? I would remove it unless there's a strong case for adding this check (and even then maybe it should be a separate PR since this one focuses on embeddings)